client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -21,110 +22,142 @@ const siteTypeOptions: Record< SiteType, string > = {
 };
 
 export default function OnboardingEdit() {
-	const [ verticalId, setVerticalId ] = useState< string >();
-
-	const { siteTitle, siteType } = useSelect( select => select( STORE_KEY ).getState() );
+	const [ lastSelectedQuestion, setLastSelectedQuestion ] = useState( null );
 
-	// This is a difficult section for working with FormToken field. Not ideal for our use case.
-	const [ verticalLabels, verticalLabelToId ] = useSelect( select => {
-		const vs = select( STORE_KEY ).getVerticals();
-		if ( ! vs ) {
-			return [ [], {} ];
-		}
-		return vs.reduce< [ string[], Record< string, string > ] >(
-			( [ labels, labelToId ], v ) => [
-				labels.concat( v.vertical_name ),
-				Object.assign( labelToId, { [ v.vertical_name ]: v.vertical_id } ),
-			],
-			[ [], {} ]
-		);
-	} );
-	const verticalLabel = useSelect(
-		select => {
-			if ( ! verticalId ) {
-				return [];
-			}
-			const v = select( STORE_KEY ).getVerticalItem( verticalId );
-			return v ? [ v.vertical_name ] : [];
-		},
-		[ verticalId ]
+	const { siteTitle, siteType, siteVertical } = useSelect( select =>
+		select( STORE_KEY ).getState()
 	);
-	const updateVerticalId = useCallback(
-		( [ nextLabel ] ) => setVerticalId( verticalLabelToId[ nextLabel ] ),
-		[ setVerticalId, verticalLabels, verticalLabelToId ]
-	);
-	// END: FormTokenField difficult section
 
-	const { resetSiteType, setSiteType, setSiteTitle } = useDispatch( STORE_KEY );
+	const { setSiteType, setSiteTitle } = useDispatch( STORE_KEY );
 
 	const updateTitle = useCallback(
 		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteTitle( e.target.value ),
 		[ setSiteTitle ]
 	);
+
 	const updateSiteType = useCallback(
 		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteType( e.target.value as SiteType ),
 		[ setSiteType ]
 	);
 
+	const selectSiteType = e => {
+		updateSiteType( e );
+		setLastSelectedQuestion( null );
+	};
+
+	const openSiteType = () => {
+		setLastSelectedQuestion( 'siteType' );
+	};
+
+	const selectSiteVertical = () => {
+		setLastSelectedQuestion( 'siteVertical' );
+	};
+
+	const selectSiteTitle = () => {
+		setLastSelectedQuestion( 'siteTitle' );
+	};
+
+	const isSelected = questionType => {
+		if ( lastSelectedQuestion ) {
+			return questionType === lastSelectedQuestion;
+		} else if ( isEmpty( siteType ) ) {
+			return questionType === 'siteType';
+		} else if ( isEmpty( siteVertical ) ) {
```

Comment:
Should use 
```suggestion
		} else if ( ! isFilledFormValue( siteVertical ) ) {
```

(same for all other instanes of `isEmpty( siteVertical )`)